lib/commands/project/project.js

```diff
@@ -363,3 +373,57 @@ function spawn(command, options, cwd, cb) {
     cb(err);
   });
 }
+
+
+//.option('-p, --path-name [path]', 'a sepecific path of the api')
+//.option('-f, --test-module <module>', 'one of: ' + testmodules)
+//.option('-t, --assertion-format <type>', 'one of: ' + assertiontypes)
+function testGenerate(directory, options, cb) {
+  var pathList = [];
+  var desiredPaths = [];
+  var swaggerSpec = require('../../util/spec');
+
+  //read the yaml file and validate it
+  readProject(directory, options, function(err, project) {
+    if (err) { return cb(err); }
+    swaggerSpec.validateSwagger(project.api.swagger, options, cb);
+
+    // get the array of string paths from json object
+    pathList = Object.keys(project.api.swagger.paths);
+
+    //check if the test frame is one of the two
+    if (options.testModule && !_.includes(TESTMODULES, options.testModule)) {
+      return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.testModule, TESTMODULES.join(', '))));
+    }
+
+    // check if the assertion-format is one of the three
+    if (options.assertionFormat && !_.includes(ASSERTIONTYPES, options.assertionFormat)) {
+      return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.assertionFormat, ASSERTIONTYPES.join(', '))));
+    }
+
+    // process the paths option
+    if (options.pathName){
+      var reg = new RegExp(options.pathName);
+      desiredPaths = pathList.filter(function(val) {
+        return val.match(reg);
+      });
+    }
+
+    // get the result
+    var config = {
+      pathName: desiredPaths,
+      testModule: options.testModule,
+      assertionFormat: options.assertionFormat
+    };
+    var result = template.testGen(project.api.swagger, config);
+
+    //output the result
+    async.series(result, function(item, cb) {
+      var path = item.name;
+      var filename = path.substirng(1) + '.js';
+      fs.writeFile('./test/' + filename, item.test, function (err) {
```

Comment:
Use the path module for putting together strings to make a path

@theganyo can you help us with path structure?